### ISSUE TITLE
Thread crashes on android when going home page

### ISSUE THREAD

**Issue by kevinresol** [underscorediscovery/snow#79]
```haxe 
var t = Thread.create(function() { } )
```
Run the above code in `ready()` and then press the android "home button", the app will crash with error:

`hxcpp: GetLocalAllocMT - requesting memory from unregistered thread!`

or it should be reported in hxcpp?

**Comment by underscorediscovery** [comment]
Some tips so long.
Rebuild snow in debug:
`cd <snow>/project/`
`flow build android --debug --d snow_dynamic_link`

Ensure your app is in debug.
Ensure ART jni debugging is enabled https://source.android.com/devices/tech/dalvik/

For other information:
snow doesn't create threads,
snow doesn't jump back and forth between java/haxe (only SDL layer does)
we need more information to debug the callstack - the best option is to have it crash with a proper callstack (i.e the debug snow should help a lot at first)

**Comment by kevinresol** [comment]
Is this the supposed output?
![capture](https://cloud.githubusercontent.com/assets/1103788/11216530/9b2c86d4-8d86-11e5-9927-96c066ecd304.PNG)

**Comment by underscorediscovery** [comment]
With a debug snow build the `libsnow` lines would show actual cpp locations

**Comment by kevinresol** [comment]
oops right, I missed something, let me try again...

**Comment by kevinresol** [comment]
Doesn't build on my Windows machine, what are the prerequisites?
![capture](https://cloud.githubusercontent.com/assets/1103788/11216686/90700bb6-8d87-11e5-9cdb-51a7751fa95c.PNG)

**Comment by kevinresol** [comment]
Here it is:
![capture](https://cloud.githubusercontent.com/assets/1103788/11217138/c12c3886-8d89-11e5-9e93-f5a75ff0610b.PNG)

**Comment by underscorediscovery** [comment]
That's much better. 

It appears that the `SDL_AddEventWatch` is pushing a message when another thread is active. 
Knowing where and why is what was needed to fix it or work around it - I'll try a fix soon (but with dev log 8 you'll see a lot of changes are headed in here, which might happen first).
